tests/baselines/reference/bestCommonTypeOfTuple2.types

```diff
@@ -29,7 +29,7 @@ class F extends C { f }
 class C1 implements base1 { i = "foo"; c }
 >C1 : C1
 >base1 : base1
->i : string
+>i : any

```

Comment:
I think this is the example @JsonFreeman is thinking of:

```ts
declare function f<T>(x: T, y: T): T;

const a = f(100, undefined);
const b = f(undefined, 200);
```

What will happen here? What will we infer for the types of `a` and `b`?